src/unix/core.c

```diff
@@ -983,3 +984,49 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_homedir(char* buffer, size_t* size) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  size_t bufsize;
+  size_t len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+
+  if (bufsize <= 0)
+    return -EIO;
+
+  buf = malloc(bufsize);
+
+  if (buf == NULL)
+    return -ENOMEM;
+
+  r = getpwuid_r(getuid(), &pw, buf, bufsize, &result);
```

Comment:
Sorry I thought the comment about retrying was specific to `getpwnam_r()`. I'll update this.